dd-sdk-android-core/src/main/kotlin/com/datadog/android/v2/api/InternalLogger.kt

```diff
@@ -11,6 +11,29 @@ import com.datadog.tools.annotation.NoOpImplementation
 
 /**
  * A Logger used to log messages from the internal implementation of the Datadog SDKs.
+ *
+ * Rule of thumb to decide which level and target we're using for the Internal Logger usage:
+ *
+ * - Target.USER: the message needs to either be actionable or provide information about the main
+ *     steps in data processing (tracking, storage, upload).
+ *     - Level.ERROR: for any actionable error originated from a user's configuration, preventing
+ *         a feature from working, or for an issue resulting in unexpected data loss;
+ *     - Level.WARN: to inform of an actionable misconfiguration or missuses of the SDK, resulting
+ *         in delayed or incomplete data;
+ *     - Level.INFO: information about important expected event (e.g.: successful upload);
+ * - Target.TELEMETRY: any event that need to be tracked for usage monitoring or for error
+ *     diagnostic.
+ *     - Level.ERROR, Level.WARN: for any critical error that is unexpected enough and actionable;
+ *     - Level.INFO, Level.DEBUG, Level.VERBOSE: important information about critical parts of the
+ *         SDK we want to monitor;
+ * - Target.MAINTAINER: can be anything relevant about the moving parts of the core SDK or any
```

Comment:
I was always curious if we really need `MAINTAINER` target, but I guess even if we remove it, some errors which are for `MAINTAINER` target are not really actionable on the user side if logged with `USER` target. But at the same time some things logged to `MAINTAINER` can be also useful to see on the `USER` target, so that they give more context for the debugging.

Anyway, this is just some thoughts, we can keep it for now.